portal/server/helpers/pageSpecificValidationErrors.js

```diff
@@ -4,126 +4,115 @@
+ }
+
+ return false;
 };
 
 const mapRequiredValidationErrors = (validationErrors, fields) => {
-  const mappedErrors = validationErrors;
-  const allRequiredFields = requiredFieldsArray(fields);
+ const mappedErrors = validationErrors;
+ const allRequiredFields = requiredFieldsArray(fields);
 
-  mappedErrors.errorList = filterErrorList(validationErrors.errorList, allRequiredFields);
+ mappedErrors.errorList = filterErrorList(validationErrors.errorList,
+allRequiredFields);
 
-  return {
-    ...generateErrorSummary(
-      mappedErrors,
-      errorHref,
-    ),
-  };
+ return {
+   ...generateErrorSummary(mappedErrors, errorHref),
+ };
 };
 
 /*
-  ALWAYS_SHOW_ERROR_FIELDS are fields that are:
-  - not required (therefore optional)
-  - BUT if this field has an error, always show it after form submit.
-  - this is the opposite of 'page specific validation rules'
-  - this is currently only used for Companies House validation.
+ ALWAYS_SHOW_ERROR_FIELDS are fields that are:
+ - not required (therefore optional)
+ - BUT if this field has an error, always show it after form submit.
+ - this is the opposite of 'page specific validation rules'
+ - this is currently only used for Companies House validation.
 */
 const hasSubmittedAlwaysShowErrorFields = (allFields, submittedFields) => {
-  const { ALWAYS_SHOW_ERROR_FIELDS } = allFields;
+ const { ALWAYS_SHOW_ERROR_FIELDS } = allFields;
 
   const hasAlwaysShowFields = ALWAYS_SHOW_ERROR_FIELDS && ALWAYS_SHOW_ERROR_FIELDS.length > 0;
 
-  if (hasAlwaysShowFields) {
+ if (hasAlwaysShowFields) {
     const pageFields = Object.keys(submittedFields).filter((fieldName) =>
       ALWAYS_SHOW_ERROR_FIELDS.includes(fieldName));
 
-    if (pageFields.length > 0) {
-      return true;
-    }
-  }
+   if (pageFields.length > 0) {
+     return true;
+   }
+ }
 
-  return false;
+ return false;
 };
 
 const mapAlwaysShowErrorFields = (validationErrors, fields) => {
-  const mappedErrors = validationErrors;
+ const mappedErrors = { ...validationErrors };
 
-  mappedErrors.errorList = filterErrorList(validationErrors.errorList, fields.ALWAYS_SHOW_ERROR_FIELDS);
+ const filteredErrorList =
+filterErrorList(validationErrors.errorList,
+fields.ALWAYS_SHOW_ERROR_FIELDS);
 
-  return {
-    ...generateErrorSummary(
-      mappedErrors,
-      errorHref,
-    ),
-  };
+ return {
+   ...generateErrorSummary({ ...mappedErrors, errorList:
+filteredErrorList }, errorHref),
+ };
 };
 
 const mapRequiredAndAlwaysShowErrorFields = (validationErrors, allFields) => {
-  const mappedErrors = validationErrors;
-  const allRequiredFields = requiredFieldsArray(allFields);
-  const alwaysShowErrorFields = allFields.ALWAYS_SHOW_ERROR_FIELDS;
-
-  const fieldsToReturn = [
-    ...allRequiredFields,
-    ...alwaysShowErrorFields,
-  ];
-
-  mappedErrors.errorList = filterErrorList(validationErrors.errorList, fieldsToReturn);
-
-  return {
-    ...generateErrorSummary(
-      mappedErrors,
-      errorHref,
-    ),
-  };
-};
```

Comment:
Could we re-add all the deleted line spaces as it makes the code more readable please? ie line spaces between different functions

Reply:
Done